lib/sqlalchemy/connectors/pyodbc.py

```diff
@@ -77,11 +77,17 @@ def create_connect_args(self, url: URL) -> ConnectArgsType:
         else:
 
             def check_quote(token: str) -> str:
-                if ";" in str(token) or str(token).startswith("{"):
+                # token is already quoted, so return it as-is
+                if token.startswith("{") and token.endswith("}"):
+                    return token
+                if ";" in token or token.startswith("{"):
                     token = "{%s}" % token.replace("}", "}}")
                 return token
 
-            keys = dict((k, check_quote(v)) for k, v in keys.items())
+            keys = dict(
+                (k, check_quote(v) if type(v) == str else v)
```

Comment:
Oops; I didn't intend to push this change - it's something I'd been experimenting with locally.  I'm not certain whether an equality check against the `str` type is reliable / universal.